packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -295,14 +295,29 @@ export class TransactionAnalyzer {
   /**
    * Calculates and returns the fee rate required for a successful CPFP.
    * @returns {string} The CPFP fee rate in satoshis per vbyte
+   * @throws {Error} When CPFP is not needed or not possible
    */
   get cpfpFeeRate(): string {
+    // Sanity check because we never know
+    if (this.estimatedCPFPChildSize <= 0) {
+      throw new Error("Invalid CPFP child size: cannot be zero or negative");
+    }
     const desiredPackageFee = new BigNumber(this.targetFeeRate).multipliedBy(
       this.CPFPPackageSize,
     );
-    const expectedFeeRate = BigNumber.max(
-      desiredPackageFee.minus(this.fee).dividedBy(this.estimatedCPFPChildSize),
-      new BigNumber(0),
+    const feeDifference = desiredPackageFee.minus(this.fee);
+
+    // Check if the parent transaction already meets or exceeds the target fee rate , this was added because of a near impossible edge cause
+    // Discussed in this thread : https://discord.com/channels/1352754651044515941/1352758476870516796/1417181497302974484
```

Comment:
Let's not link to discord as I'm not sure how permanent that is. try and explain the edge case in a code comment if necessary to clarify the code.